rust/src/tx_builder.rs

```diff
@@ -140,7 +143,33 @@ fn fake_full_tx(tx_builder: &TransactionBuilder, body: TransactionBody) -> Resul
     })
 }
 
+fn assert_required_mint_scripts(mint: &Mint, maybe_mint_scripts: Option<&NativeScripts>) -> Result<(), JsError> {
+    if maybe_mint_scripts.is_none_or_empty() {
+        return Err(JsError::from_str(
+            "Mint is present in the builder, but witness scripts are not provided!",
+        ));
+    }
+    let mint_scripts = maybe_mint_scripts.unwrap();
+    let witness_hashes: HashSet<ScriptHash> = mint_scripts.0.iter().map(|script| {
+        script.hash(ScriptHashNamespace::NativeScript)
```

Comment:
It's possible this will be a bottleneck as `fee_for_input()` (which in turn calls `min_fee()` and ultimately this) is called over and over when adding inputs or considering it.

Of course we can always profile this later as part of an entire typical workflow to actually see since this is just my guess based on the last time I profiled the builder, which didn't use automatic input selection. I'm not 100% sure if the hashing was as bad as the key conversion operations either. And if it is a bottleneck we can just make a new PR since it won't even affect major version number or anything. Avoiding it might make code complicated too so I think it's totally fine to just merge this PR with the current behavior. I just wanted to document this now in this review so I don't forget later.

Reply:
Good point. I have commented out the check in `min_fee` for now as it is not critical and was mostly added for extra assurance. Added a comment there that it can be reworked to cache hashing or maybe only perform the check once if the mint has not changed since, but not enough time to work on this atm